Merge master

jfsiii opened the pull request:
## Summary

Merge in master to keep this feature branch current.

Specifically hoping https://github.com/elastic/kibana/pull/38184 will address the [CI failure](https://github.com/elastic/kibana/pull/37950/checks?check_run_id=142851242) in #37950 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- [ ] ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [ ] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [ ] ~~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
- [ ] ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] ~~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
- [ ] ~~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

dmlemeshko committed:
skip Code History functional tests on Firefox (#37954)

weltenwort committed:
[Logs UI] Include correct fields in "entries after" query (#37881)

This fixes #37704 by correcting the fields list used in the second part of the `getLogEntriesBetween` library function.

jgowdyelastic committed:
[ML] Reverting incorrectly removed scope parameters (#37873)

* [ML] Reverting incorrectly removed scope parameters

* removing  no-unused-vars comments

jgowdyelastic committed:
[Watcher] Reverting incorrectly removed scope parameters (#37875)

* [Watcher] Reverting incorrectly removed scope parameters

* removing  no-unused-vars comments

maryia-lapata committed:
[Vis: Default editor] EUIficate IP Ranges (#36896)

* Create IpRangeType and IpRanges controls

* Add validation

* Refactoring

* Add behavior when discarding changes

* Refactoring: create common input list

* Remove old template

* Move add btn to input_list, add placeholder

* Remove unused directives

* Remove unused translations

* Refactoring

* Use EuiButtonGroup instead of toggle button

* Update options ids, add aria-labels

* Remove unused translations

* Update mask model, update TS, update aria labels

* Add validation for CIRD mask

legrego committed:
Security - display message when logging out of kibana (#37234)

* display message when logging out of kibana

* update provider tests

* fix token logout test

* remove newline

legrego committed:
Feature Controls - update base privilege tooltip (#37868)

legrego committed:
Feature Controls - automatically grant access to short urls (#37532)

* automatically grant access to short urls

* adds API integration tests

* Update x-pack/test/api_integration/apis/short_urls/index.ts

Co-Authored-By: Brandon Kobel <brandon.kobel@gmail.com>

Bamieh committed:
[i18n] [docs] Update docs to encourage static i18n import (#37858)

* update documentation

* Update docs/development/plugin/development-plugin-localization.asciidoc

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/README.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/README.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

jkakavas committed:
Add section about OpenID Connect in docs (#37525)

justinkambic committed:
Add functional test to cover heartbeat location data. (#37891)

XavierM committed:
Fix bug when you delete column (#37946)

mattkime committed:
remove unneeded eslint-disable directives (#37922)

* remove unneeded eslint-disable directives

ryankeairns committed:
[Canvas] Fix workpad list overflow (#37345)

* fix workpad list overflow

* make file picker full width on small devices

* fix modal height, add comments

cjcenizal committed:
Add repository-azure autocompletion settings to Console (#37935)

Add repository-azure as an option when creating a repository.

Commit:
Add 1/2 second sleep to fix test (#37903)

spong committed:
[SIEM] Fix IP Overview query to query for most recent Host (#37723)

## Summary

As outlined in https://github.com/elastic/ingest-dev/issues/453, this PR fixes the `IP Overview` query to query for the most recent Host.

Now the Host Data shown will be the most recent host for the given IP regardless of whether `Source` or `Destination` is selected in the FlowTargetSelect.

![image](https://user-images.githubusercontent.com/2946766/58731558-46fb5200-83ac-11e9-97d5-7db5dd8ac84e.png)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

Commit:
[SIEM] Host Details Page Updates (#37769)

* toggle to exclude host ID links on host details pg

* first attempt at hiding host column

* switch to conditionally splicing array of columns

* exclude dest from auth table on host details pg

* minor overview style touchups

* rm some extranous EuiFlexGroup; more need removal

* snapshots update

* Revert "snapshots update"

This reverts commit bdd6b8e3165b7488ade99a349d1a181f4abd17f9.

* update siem snapshots

* remove unused arg

* change uncommon process column order per #503

* simplify if condition, per steph

* update tests, per steph

* simplify `hostIdRenderer` per xavier

* update snapshots

* fix/improve unit tests, per xavier

* Revert "update snapshots"

This reverts commit dc0f79606bb508c657cf380eaa9028f1c3e0c845.

* add host links in uncommon processes table

* update uncommon process test, thx to xavier

FrankHassanabad committed:
[SIEM] Authentication fixes for winlogbeat and ordering (#37718)

## Summary
  * Enabled winlogbeat to work with workaround for authentication_failed (until that is fixed)
  * Fixed ordering to be successes first, failures second
  * https://github.com/elastic/ingest-dev/issues/491
  * https://github.com/elastic/ingest-dev/issues/469

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

zfy0701 committed:
[Code]: add ui.enabled flag and fix uneccessary usage of enabled flag (#38018)

chandlerprall committed:
Make plugins/reporting config snapshot deterministic (#38019)

cqliu1 committed:
[Canvas] Fixes broken elements in web logs sample data workpad (#37931)

* Fixed broken elements in logs sample workpad. Cleaned up expression formatting in each element

* Removed return characters at the beginning of expressions

kobelb committed:
Fixing isMigrated bug, when migrations failed we incorrectly ignored it. (#37674)

Prior to this change, when migrations failed (for example when a
.kibana_1 index already exists and no alias does) the awaitMigration
promise would resolve incorrectly. This allowed subsequent plugins,
including Spaces, to proceed using the SavedObjectsClient/Repository
to create documents. This commonly caused a .kibana index to be created,
which would then block the creation of a .kibana alias, and everything
would grind to a halt.

mw-ding committed:
[Code] update percentage (#38003)

mw-ding committed:
[Code] Filter out binary files for indexer (#37945)

lukeelmers committed:
Fix bad type import from interpreter. (#37741)

mw-ding committed:
[Code] use empty placeholder when search filters returns empty results (#38006)

igoristic committed:
Optimized content to be more vertical for resizing consistancy (#37120)

* Optimized content to be more vertical for resizing consistancy

* Fixed snapshots

* Feedback changes

* Update _summary_status.scss

* Fixed column order for testing

Commit:
[Code] fix hover in monaco-editor 0.17 (#37964)

zfy0701 committed:
[Code]: Use more robust way to handle url (#38060)

Commit:
[Code] fix url and encoding for java langserver on Windows (#38064)

Commit:
fix(code/frontend): catch structure generating error in saga (#37957)

flash1293 committed:
Add error handling to expression runner (#37968)

Commit:
fix(code/frontend): improve regexp that matches the container of a symbol (#37856)

Commit:
fix(code/frontend): should not polling status if import repo not succeed (#37955)

simitt committed:
[apm] update index pattern (#37866)

related to https://github.com/elastic/apm-server/pull/2239

peteharverson committed:
[ML] Fixes loading of Single Metric Viewer if partition field is text (#37975)

* [ML] Fixes loading of Single Metric Viewer if partition field is text

* [ML] Fix failing test and edit translations for cardinality check

* [ML] Edits to fields_service.js following review

mikecote committed:
Fix export saved objects API docs (#37991)

mikecote committed:
Update import saved objects API docs to use curl examples (#37997)

* Remove copy as curl

* Update import docs to use curl examples

justinkambic committed:
Remove deprecated feature from elasticsearch queries. (#37995)

justinkambic committed:
[Uptime] [Bug] Disable integration links for apps that are not present (#37284)

* Add tests and checks for disabled/enabled integration link.

* Add empty state if all integrations are unavailable.

* Fix broken type.

jgowdyelastic committed:
[ML] Fix for constant polling of deleting job tasks (#38013)

* [ML] Fix for constant polling of deleting job tasks

* fixing comment

mikecote committed:
Fix URLs from displaying too long in saved objects management UI (#37987)

* Fix URLs from displaying too long

* Add encodeURIComponent

kobelb committed:
Updating mixin-deep to 1.3.1 (#38057)

mistic committed:
Relative cache loader (#36131)

* chore(NA): first efforts to enable cache-loader relative paths and disable babel-register cache. chore(NA): allow to use cache-loader results everywhere as a persistent cache.

* chore(NA): last change to support cache loader across envs.

* fix(NA): cache-loader config generator function scope.

* refact(NA): static cache config for dll compiler.

* chore(NA): support for force writable cache flag. use built assets for cache on distributable.

* chore(NA): update cache-loader to last version. fix(NA): logic bug with force writable cache flag.

* chore(NA): move optimize cache to built_assets.

* chore(NA): relative paths for sourcemap configurations.

* chore(NA): remove redudant check for process.env var.

* refact(NA): simplify double negated expression.

* chore(NA): remove webpack prefix from devtoolModuleFilenameTemplate.

* chore(NA): rename cache write force flag to KBN_CACHE_LOADER_WRITABLE.

* chore(NA): readd babel register cache.

* chore(NA): change .cache to .ui_bundles_cache

* chore(NA): change .ui_bundles_cache to cache/ui_bundles

* chore(NA): update gitignore with .babel_register_cache.json

* chore(NA): rollback devtool module filename template to asbolute resource path.

* chore(NA): apply cache-loader patch.

clintandrewhall committed:
[Canvas] Interpreter type conversion (#37300)

* Move Expression types to OSS.

* Add interfaces for KibanaContext and KibanaDatatable types.

* Convert esaggs expression function to TypeScript.

* Clean up KibanaDatatable and KibanaContext types.

* Convert functions to use new Interpreter types

* Update ArgumentString-related types to simply be TypeString.

* Rename types.ts to index.ts so types can be imported at the top level.

* Update ArgumentString-related types to simply be TypeString.

* Rename types.ts to index.ts so types can be imported at the top level.

* Remove unused code from arguments types.

* Convert x-pack imports of OSS plugins from relative to absolute.

* Fix esaggs import.

* Addressing self feedback

ryankeairns committed:
default shape aspect ratio to false (#38030)

dmlemeshko committed:
functional tests/firefox wait for loading after page refresh (#38082)

nickpeihl committed:
Use production EMS manifest (#38050)

vitalics committed:
[TSVB] [AT] implement new test cases for time series formatter (#37506)

* implement new test cases for time series formatter

justinkambic committed:
Remove "common" options from SDP that have a 0ms difference between end and start time. (#37395)

lukeelmers committed:
Fix broken filtering on metric visualizations. (#38029)

Bamieh committed:
Allow headers config to propagate in `callAsCurrentUser` (#34610)

* add telemetry query source when hitting monitoring indexes

* allow headers to pass through

* Update src/core/server/elasticsearch/scoped_cluster_client.ts

* fixes from bill code review

* split test cases

* fix functional test

* check if issue from intersect function

* more debug

* check for duplicate values instead

* code review updates

* remove mockClear

nreese committed:
[Maps] replace FormattedText with FormattedMessage (#38016)

* [Maps] replace FormattedText with FormattedMessage

* fix Capitalization

Commit:
Move shimming of workpad CSS to only happen on fetch (#38117)

Commit:
Fix discover not using an index pattern when 2 are created (#38053)

zfy0701 committed:
[Code]: unassociate html with typescript langserver (#38150)

cjcenizal committed:
Skip Index Manamgent API integration tests that are inapplicable to Cloud, when they are run on Cloud. (#38045)

kobelb committed:
Updating stringstream to 0.0.6 (#38155)

lukeelmers committed:
ui settings mocks: mock observables' return values. (#37584)

joshdover committed:
[new-platform] Simplify the frontend integration with the legacy platform (#37734)

TinaHeiligers committed:
KQL in TSVB (#36784)

* Squashed commit of the following:

commit 0cff8243a99ff23421c3cbb15d62b26332a31e21
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Mon May 20 16:16:15 2019 -0700

    Adds ignore_lobal_filter YesNo component back

    Deletes fixtures file

commit e54e1690c9c46e2de5e747d14030a7e27e74c138
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Mon May 20 14:39:21 2019 -0700

    squash-merges adding KQL with the full QueryBar and SavedObject migrations

Debugging

Adds filter defaults

Updates index pattern for the QueryBar on componentDidUpdate

squash-merges adding KQL with the full QueryBar and SavedObject migrations required

Adds ignore_lobal_filter YesNo component back

Deletes fixtures file

Updates migration test number

* Changes QueryBar to QueryBarInput in TimeseriesPanelConfigUi

Replaces QueryBar with QueryBarInput in Timeseries

Changes QueryBar to QueryBarInput in TSVB Metric

Changes QueryBar to QueryBarInput in other TSVB visualisation editors

Removes async from componentWillMount

changes indexPattern to indexPatternObject in series and table query request processors

* Squashed commit of the following:

commit 2c3107963cff20b2d24323fddfaf80c848b86848
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 15:15:06 2019 -0700

    Cleans up comments

commit 943150d4b9b71183e9d5b26c95449ea4519aad4c
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 14:09:53 2019 -0700

    Cleans up parser check more

commit d31de88525595720556440ae71515f5772774eaa
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 14:01:02 2019 -0700

    Cleans up parser check

commit b592071b1d939ec2df9b9c921d57a22dac54e703
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 13:16:55 2019 -0700

    skips server call on invalid kuery queries

Cleans up

Deletes duplicate migration function

Adds 's' to split_by_filters method

Resolves issues after rebase

* Removes quotes around index pattern strings passed into fetch_index_patterns, adds comments on reasoning

* queryBar to queryBarInput

Adds transformFilterStringToQueryObject back to visualizations migrations that was lost during rebase

* Deletes unused code

* Fixes migrations test

* compiles visualizations migrations for version 7.3.0

* temporary change to query_bar_input fetch_index_patterns file

formatting

Includes code from PR#37413

Clean up

* Formatting

* Uses series index pattern when override_index_pattern is selected and an index pattern is provided

* Removes empty object as alternative to localStorage

* Removes handelQueryChange from where it is not needed

* extracts retrieval of the default query language from uiSettings into a helper method

* Resolves some PR comments

* Handles Query Error from an invalid syntax query

* Converts string queries into objects with kuery as the language in saved_objects data for ecommerce, flights and logs TSVB visualizations

* Deletes unused translation items

* Adds optional id prop to the query_bar_input and makes the screenTitle optional

* Updates [eCommerce] Sold Products per Day

* Removes screenTitle from component QueryBarInput

* Wraps index pattern strings in a double quote to allow for names such as 'logstash-*'

mw-ding committed:
[Code] search functional test to cover file typeahead query case insensitive (#38012)

mw-ding committed:
[Code] Improve the document search highlights for non-english query (#38154)

jen-huang committed:
Fix i18n issues (#38169)

MichaelMarcialis committed:
[SIEM] Beta badge on overview page (#38161)

* add beta badge to overview header

* update test

* updates snapshot

nickpeihl committed:
Fix sample data functional test (#38184)

Commit:
[Code] fix another path problem on windows (#38072)

stephmilovic committed:
[SIEM] Additional Overview Network & Hosts metrics (#38005)

[SIEM] Additional Overview Network & Hosts metrics (#38005)

mistic committed:
chore(NA): declare kbn-babel-preset as peer dependency for kbn plugin helpers. (#37710)

clintandrewhall committed:
[Canvas] Localize Function Errors (#37381)

* [Canvas] Localize Function Errors

* Update test

* Fix enum usage

Commit:
[Code] improve setup guide (#38218)

FrankHassanabad committed:
[SIEM] Changes out event.severity for message in timeline and fixes sorting bug (#38158)

## Summary

  * Change out event.severity for message in timeline
  * Fixed critical crash where you should not be able to sort on timeline columns that are not aggregatable
  * https://github.com/elastic/ingest-dev/issues/513
  * https://github.com/elastic/ingest-dev/issues/496

Error toaster fixes:
<img width="348" alt="Screen Shot 2019-06-05 at 1 59 01 PM" src="https://user-images.githubusercontent.com/1151048/58986428-97065a00-879a-11e9-8ff9-003ac29f6d9e.png">

What the timeline looks like with severity replaced with message:
<img width="883" alt="Screen Shot 2019-06-05 at 1 55 19 PM" src="https://user-images.githubusercontent.com/1151048/58986459-a71e3980-879a-11e9-8144-f16f055bd53d.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

spong committed:
[SIEM] Removing draggability from message field (#38166)

## Summary

Currently, searching the `message` field within Timeline will not function properly, so in effort to prevent the user from searching with it, this PR removes the areas where `message` is draggable, which includes:

* Host Events Table
* Host Details Events Table
* Timeline Renderers
* Timeline Event Details

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

andrew-goldstein committed:
## [SIEM] Disables Timeline Provider Mutations While Loading Data (#38185)

## Summary

* Fix: disables dropping a new data provider provider on the timeline, deleting an existing data provider, or choosing an action from the data provider context menu while the timeline is loading data to prevent draggable errors, like the one pictured in the screenshot below:

![draggable-error](https://user-images.githubusercontent.com/4459398/58996789-ebb7ce00-87b6-11e9-9490-47a0df6f89ab.png)

* Fix: the timeline settings icon appears too close to the `Refresh` button:
### Before (Chrome)

![timeline-gear-before](https://user-images.githubusercontent.com/4459398/58997030-e6a74e80-87b7-11e9-86df-cb67592e63a7.png)

### After (Chrome)

<img width="1679" alt="timeline-gear-after-chrome" src="https://user-images.githubusercontent.com/4459398/58997064-1ce4ce00-87b8-11e9-9ffb-0f412e0018ad.png">

### After (Firefox)

<img width="1672" alt="timeline-gear-after-firefox" src="https://user-images.githubusercontent.com/4459398/58997113-4f8ec680-87b8-11e9-9b52-dc8a13d51361.png">

### After (Safari)

<img width="1680" alt="timeline-gear-after-safari" src="https://user-images.githubusercontent.com/4459398/58997169-911f7180-87b8-11e9-8186-0ff31e41ad53.png">

### After (IE 11)

<img width="1354" alt="timeline-gear-after-ie11" src="https://user-images.githubusercontent.com/4459398/58997306-40f4df00-87b9-11e9-844d-b12c7e16f7cb.png">

* Fix: the `Open Timeline` modal is showing bulk-delete actions

### Before (Chrome)

![open-timeline-before](https://user-images.githubusercontent.com/4459398/58997349-700b5080-87b9-11e9-8299-7dbeccbd8ff9.png)

### After (Chrome)

<img width="1008" alt="open-timeline-after" src="https://user-images.githubusercontent.com/4459398/58997381-9af5a480-87b9-11e9-9909-2d68bd5ff2c3.png">

* Fix: action icons on the `Timelines` page and `Open Timelines` modal are not aligned with the counts

### Before (Chrome)

<img width="1552" alt="all-timelines-before" src="https://user-images.githubusercontent.com/4459398/58997554-659d8680-87ba-11e9-868c-d92ad72f44b8.png">

### After (Chrome)

<img width="1550" alt="all-timelines-after" src="https://user-images.githubusercontent.com/4459398/58997656-d8a6fd00-87ba-11e9-9038-c0ca464f8c4d.png">

* Fix: Changed the title text of the fields browser from `Select Fields` to `Customize Columns`

### Before (Chrome)

<img width="910" alt="fields-browser-before" src="https://user-images.githubusercontent.com/4459398/58997616-a4334100-87ba-11e9-9cd8-02ed9d265bf2.png">

### After (Chrome)

<img width="910" alt="fields-browser-after" src="https://user-images.githubusercontent.com/4459398/58997691-fa07e900-87ba-11e9-8cd9-6412d2cef8f0.png">

https://github.com/elastic/ingest-dev/issues/471

ogupte committed:
[APM] Moves the APM index creation from server startup (#37965)

* [APM] Closes #37499 by moving the APM index creation from server startup
to savedObject request for the index pattern. It first checks if the
index pattern is saved, if not it creates it.

* [APM] provide more meaninful status codes in the default error handler

rudolf committed:
[Core] Rewrite saved objects in typescript (#36829)

* Convert simple files to TS

* Fix jest tests

* Rename saved_objects_client{.js => .ts}

* WIP saved_objects_client

* saved_objects repository{.js => .ts}

* includedFields support string[] for type paramater

* Repository/saved_objects_client -> TS

* Fix tests and dependencies

* Fix saved objects type errors and simplify

* saved_objects/index saved_objects/service/index -> ts

* Fix saved objects export test after switching to typed mock

* Workaround type error

* Revert "Workaround type error"

This reverts commit de3252267eb2e6bf56a5584d271b55a7afdc1c53.

* Correctly type Server.savedObjects.SaveObjectsClient constructor

* saved_objects/service/lib/index.{js -> ts}

* saved_objects/service/lib/scoped_client_provider{js -> ts}

* Typescriptify scoped_client_provider

* Fix x-pack jest imports

* Add lodash/internal/toPath typings to xpath

* Introduce SavedObjectsClientContract

We need a way to specify that injected dependencies should adhere to the
SavedObjectsClient "contract". We can't use the SavedObjectsClient class
itself since it contains the private _repository property which in TS is
included in the type signature of a class.

* Cleanup and simplify types

* Fix repository#delete should return {}

* Add SavedObjects repository test for uncovered bug

Test for a bug in our previous js implementation that can lead to data
corruption and data loss.

If a bulkGet request is made where one of the objects to fetch is of a type
that isn't allowed, the returned result will include documents which have the
incorrect id and type assigned. E.g. the data of an object with id '1' is
returned with id '2'. Saving '2' will incorrectly override it's data with that
of the data of object '1'.

* SavedObject.updated_at: string and unify saved_object / serializer types

* Cleanup

* Address code review feedback

* Don't mock errors helpers in SavedObjectsClient Mock

* Address CR feedback

* CR Feedback #2

* Add kibana-platform as code owners of Saved Objects

* Better typings for SavedObjectsClient.errors

* Use unknown as default for generic type request paramater

* Bump @types/elasticsearch

* Fix types for isForbiddenError

* Bump x-pack @types/elasticsearch

XavierM committed:
fix math (#38132)

* fix math

* add unit test

* review I

Commit:
Typescript dashboard app code. (#37527)

* Typescript dashboard app code. Pulls out part of the massive embeddable API PR.

I typscripted this code as part of that PR because it helped me find errors, but really unnecessarily blew up the size of that PR. So pulling out.

* User filter types from kbn-es-query

* Address code review feedback

* Replace require with import

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/530/)

mshustov committed:
Route tags (#37344)

* expose route info in KibanaRequest

* update mocks in test

* make tags readonly, getRouteInfo is private method

* add mocks for hapi internals

* mode deepFreeze to core utils level as it env agnostic

* freeze route props

* fix typo

* add tests for route options

* fix integration tests. deep_freeze was moved under core utils

* add comments, expose public types and regenerate docs

* address comment. remove unnecessary async in route handlers

* make routeSchema optional instead of union with undefined

* @skaapgif improvements

* update docs

peteharverson committed:
[ML] Close job after forecast errors if starting from closed (#38253)

* [ML] Close job after forecast errors if starting from closed

* [ML] Pass closeJob flag to error handler if forecast ID undefined

peteharverson committed:
[ML] Only add ML links for sample data sets if full license (#38120)

* [ML] Only add ML links for sample data sets if full license

* [ML] Update following review

kobelb committed:
[docs][skip ci] Read only access for @elastic/kibana-app applications  (#35560)

* Adding read-only mode section to Discover

* No more "mode" or "badge" terminology

* Putting newline back

* Adding "Saving" section for visualize

* Adding dashboard read only access section

* Timelion gets read-only documentation

* Adding read only access section to index patterns and advanced settings

* Adding graph read only access section

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/551/)

alvarezmelissa87 committed:
revert dragselect version (#38183)

streamich committed:
fix: 🐛 stop showing spinner only on error (#38289)

Commit:
[dev/license_checker][dev/npm] reactor, ts-ify, de-grunt (#37807)

Commit:
[dev/failed_tests] try to ignore irrelevant failures (#37902)

* [dev/failed_tests] try to ignore irrelevant failures

* fix typo

kertal committed:
Undo setting of $scope.refreshInterval at dashboard (#37175)

* Add test for passing on dashboard url params to timepicker values

* Revert unnecessary setting of $scope.refreshInterval

refreshInterval is set at $scope.model, no need to set it directly at $scope like in discover and vis

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/607/)

kobelb committed:
Updating tar to 2.2.2 (#38182)

kobelb committed:
Updating underscore.string to 3.3.5 (#38180)

kobelb committed:
Updating axios to 0.18.1 (#38179)

kobelb committed:
Updating js-yaml to 3.13.1 for more dependencies (#38160)

Renovate recently updated js-yaml in
https://github.com/elastic/kibana/pull/37809. However, a few
indirect dependencies weren't upgraded to use this new version.

simianhacker committed:
[Infra UI] Fixes #37999 - Reset error when IP address changes to valid entry (#38022)

* [Infra UI] Fixes #37999 - Reset error when IP address changes to something valid

* movnig the setError above try/catch

nreese committed:
[Maps] fix tooltip text overlap and text overflow (#38271)

* [Maps] Use table to display tooltip properties

* Design cleanup

- Added dropshadow
- Cleaned SASS
- Removed unnecessary wrappers around close button and centered it
- Change loading state to be centered
- Fixed z-index so tooltip is on top of controls
- Condensed padding/spacing

* Fix overlays in IE

* revert yarn.lock changes

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/651/)

stephmilovic committed:
[SIEM] fix ipv6 bug and write tests for ip details page (#37528)

fix ipv6 bug and write tests for ip details page

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/695/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/672/)

zfy0701 committed:
[Code]: remove hard code node heapsize (#38304)

Commit:
[ci] disable firefox tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/708/)

Commit:
update yarn.lock

tsg committed:
[SIEM] Update links to docs and website (#38313)

* Update links to docs and website in the SIEM app

* Update also link text

* Capitalize solution

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/732/)

nickpeihl committed:
Add link to 7.2 ems landing page (#38284)

FrankHassanabad committed:
Fixed cyclic dependencies (#38321)

## Summary

Fixes cyclic dependencies

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/767/)

spalger commented:
Closed this since it was triggering a new PR build on every commit to master, and since Jenkins is kinda shit at aborting old builds, and it creates a lot of noise in GCP either way, in the future it'd be great if we created a branch from the point in master that we want to merge into the feature-branch before creating the pr:

```
git fetch upstream master
git checkout -b merge-master upstream/master # create a snapshot of the master branch
git push origin merge-master # create the PR from this branch
```

ruflin commented:
@spalger Because Kibana does not allows us to merge PR's (instead of squash) we need to merge locally anyways and push. Working with a PR's was the hope to also have CI run ones, but yes as master constantly keeps updating we either would have to create our branch of it and do a PR or as we decided, locally to not have to squash.
